### PR TITLE
update JuMP and Julia syntax

### PR DESCRIPTION
**Pull request by chriscoey**
@mlubin tests are failing because an expr is a :call rather than a :comparison, can you help me fix that?

### CODE REVIEW

File: src/JuGP.jl [hunk 1]
```diff
@@ -1,34 +1,32 @@
...
+    include("solver.jl")
 
-# package code goes here
-include("types.jl")
-include("operators.jl")
-include("expr.jl")
-include("solver.jl")
+    function GPModel(;solver=nothing, method=:LogSumExp)
+        m = Model(solver=GPSolver(method, solver))
+        @NLparameter(m, foo == 1) # to work around NLP resolve warning
```
Comment (mlubin) [inline src/JuGP.jl]: This can be removed

File: src/JuGP.jl [hunk 2]
```diff
@@ -1,34 +1,32 @@
...
-    haskey(m.ext,:GP) || error("setDiscrete can only be called on GP models")
-    vals = convert(Vector{Float64},collect(values))
-    gp = m.ext[:GP]::GPData
-    gp.discretevalues[x.col] = vals
-    nothing
-end
-export setDiscrete
+    export GPModel
 
+    function setDiscrete(x::Variable, values)
```
Comment (mlubin) [inline src/JuGP.jl]: I'd recommend ditching camelCase. `set_discrete` maybe.

File: src/JuGP.jl [hunk 3]
```diff
@@ -1,34 +1,32 @@
 module JuGP
+    using JuMP
```
Comment (mlubin) [inline src/JuGP.jl]: It's conventional not to indent code because of a module block.

**Review by mlubin — COMMENTED** [review]
Oh wow, I guess this is still a thing.

File: src/JuGP.jl (hunk 2 above)
Comment (chriscoey) [reply src/JuGP.jl]: jump doesn't usually use underscores. what about `setdiscretevalues` or `setvaluelist`.

File: src/JuGP.jl (hunk 1 above)
Comment (chriscoey) [reply src/JuGP.jl]: which parts? just the `@NLparameter`?

Comment (mlubin) [reply src/JuGP.jl]: Yes, this one line.

File: src/JuGP.jl (hunk 2 above)
Comment (mlubin) [reply src/JuGP.jl]: Anything but camelCase.

**Pull request merged**